overlord/assertstate/assertmgr.go

```diff
@@ -234,17 +236,30 @@ func (e *commitError) Error() string {
 	return fmt.Sprintf("cannot add some assertions to the system database:%s", strings.Join(l, "\n - "))
 }
 
+func clarifyUnsupportedFormat(err error, a asserts.Assertion) (clarifyd error) {
+	if _, ok := err.(*asserts.UnsupportedFormatError); ok {
+		return fmt.Errorf("%v assertion format is too new for this snapd, upgrade snapd/core snap", a.Ref())
+	}
+	return nil
+}
+
 // commit does a best effort of adding all the fetched assertions to the system database.
 func (f *fetcher) commit() error {
 	var errs []error
 	for _, a := range f.fetched {
 		err := f.db.Add(a)
-		if revErr, ok := err.(*asserts.RevisionError); ok {
-			if revErr.Current >= a.Revision() {
-				// be idempotent
-				// system db has already the same or newer
-				continue
+		clarified := clarifyUnsupportedFormat(err, a)
+		if asserts.IsKeptCurrent(err) {
```

Comment:
This would then become simply
```
err := f.db.Add(a)
if asserts.IsUnsupportedUpdate(err) {
        logger.Noticef("Cannot update assertion: %v", err)
        continue
}
if err != nil {
        ...
}
```
It might be nice to add such a prefix transforming the lowercase error message into a log message.